ebean-core/src/main/java/io/ebeaninternal/server/transaction/TransactionManager.java

```diff
@@ -499,7 +492,7 @@ public final ScopedTransaction externalBeginTransaction(SpiTransaction transacti
    */
   public final ScopedTransaction beginScopedTransaction(TxScope txScope) {
     txScope = initTxScope(txScope);
-    ScopedTransaction txnContainer = activeScoped();
+    ScopedTransaction txnContainer = (ScopedTransaction) inScope();
```

Comment:
is it possible (or allowed) that we will get an instance not implementing `ScopedTransaction` here?

Reply:
if a `DB.save(entityWithPersistAction)` creates an implicit transaction, we will fail here, if a callback or `@PrePersist` action tries to create a nested transaction, because implicit transactions are not nestable